### PR TITLE
test_pg_dups_trimming.py: Verifiy automatic trimming of PGLog dups

### PR DESCRIPTION
**Pull request by pdhiran**
Covered in PR:
1. Suite and Conf files for the test run
2. Initial workflow up till injection of duplicate entries into OSDs

Signed-off-by: Pawan Dhiran <pdhiran@redhat.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [x] Create a test case in Polarion reviewed and approved.
- [x] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [x] Review the automation design
- [x] Implement the test script and perform test runs
- [x] Submit PR for code review and approve
- [x] Update Polarion Test with Automation script details and update automation fields
- [x] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pdhiran** [comment]
Pass logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-WSM1CP

**Comment by pdhiran** [comment]
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-5T9W7J/

**Comment by pdhiran** [comment]
Test Case Passed log . http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-0S9LX3

File: suites/pacific/rados/tier-3_rados_test-pg-dups-trimming.yaml [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      module: rados_prep.py
+      polarion-id: CEPH-83574472
+      config:
+        email_alerts:
+          smtp_host: smtp.corp.redhat.com
+          smtp_sender: ceph-iad2-c01-lab.mgr@redhat.com
+          smtp_port: 25
+          interval: 10
+          smtp_destination:
+            - pdhiran@redhat.com
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply suites/pacific/rados/tier-3_rados_test-pg-dups-trimming.yaml:140]: Which email would you suggest to send the cluster alerts to?

File: tests/rados/test_pg_dups_trimming.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    for pool in pools:
+        pool_id = pool_obj.get_pool_id(pool_name=pool)
+        pgid = f"{pool_id}.0"
+        pg_set = rados_obj.get_pg_acting_set(pg_num=pgid)
+        acting_sets[pgid] = pg_set
+        [osds.append(osd) for osd in pg_set]
+
+    log.info(f"Identified Acting set of OSDs for the Pools. {acting_sets}")
+
+    # COllect num pglog objects from dump_mempools
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply tests/rados/test_pg_dups_trimming.py:68]: changed.

File: tests/rados/test_pg_dups_trimming.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    test_image = config.get("container_image")
+    log.debug(f"Verifying pglog dups trimming on OSDs, test img : {test_image}")
+
+    with open(pool_configs_path, "r") as fd:
+        pool_conf_file = yaml.safe_load(fd)
+
+    pools = []
+    acting_sets = {}
+    osds = []
+    for i in pool_configs.values():
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply tests/rados/test_pg_dups_trimming.py:41]: done

File: suites/pacific/rados/tier-3_rados_test-pg-dups-trimming.yaml [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        pool_configs:
+          pool-1:
+            type: replicated
+            conf: sample-pool-1
+          pool-2:
+            type: replicated
+            conf: sample-pool-2
+          pool-3:
+            type: erasure
+            conf: sample-pool-3
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply suites/pacific/rados/tier-3_rados_test-pg-dups-trimming.yaml:165]: done

File: suites/pacific/rados/tier-3_rados_test-pg-dups-trimming.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pdhiran) [reply suites/pacific/rados/tier-3_rados_test-pg-dups-trimming.yaml]: Done.

File: suites/pacific/rados/tier-3_rados_test-pg-dups-trimming.yaml [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+                - rgw.1
+              args:
+                placement:
+                  label: rgw
+
+  - test:
+      name: Configure client admin
+      desc: Configures client admin node on cluster
+      module: test_client.py
+      polarion-id:
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply suites/pacific/rados/tier-3_rados_test-pg-dups-trimming.yaml:117]: Will check and update with the next PR.

File: suites/pacific/rados/tier-3_rados_test-pg-dups-trimming.yaml [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+      polarion-id: CEPH-83574472
+      config:
+        email_alerts:
+          smtp_host: smtp.corp.redhat.com
+          smtp_sender: ceph-iad2-c01-lab.mgr@redhat.com
+          smtp_port: 25
+          interval: 10
+          smtp_destination:
+            - cephci@redhat.com
+          smtp_from_name: CM Scenarios - Dups trimming Cluster Alerts
```
Comment (pdhiran) [inline suites/pacific/rados/tier-3_rados_test-pg-dups-trimming.yaml:143]: Let me know if this from name for the mails would make sense.

**Review by psathyan — COMMENTED** [review]
Other than the missing Polarion ID... the code looks good to me.

File: suites/pacific/rados/tier-3_rados_test-pg-dups-trimming.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (pdhiran) [reply suites/pacific/rados/tier-3_rados_test-pg-dups-trimming.yaml]: Added the test ID.

**Comment by pdhiran** [comment]
> Other than the missing Polarion ID... the code looks good to me.

Added the missing polarion ID

**Review by psathyan — COMMENTED** [review]
Looks good to me

File: conf/pacific/rados/6-node-depressa-cluster.yaml [hunk 7]
```diff
@@ -0,0 +1,93 @@
+globals:
+  -
+    ceph-cluster:
```
[the remark this answers is not in the document]
Comment (pdhiran) [reply conf/pacific/rados/6-node-depressa-cluster.yaml:3]: Focusing on deploying OSDs only on NVMe disks

**Review by tintumathew10 — APPROVED** [review]
Looks good to me.

**Pull request merged**